src/operator/controllers/intents_reconcilers/linkerd_reconciler.go

```diff
@@ -0,0 +1,126 @@
+package intents_reconcilers
+
+import (
+	"context"
+	"errors"
+
+	otterizev1alpha3 "github.com/otterize/intents-operator/src/operator/api/v1alpha3"
+	"github.com/otterize/intents-operator/src/operator/controllers/intents_reconcilers/consts"
+	linkerdmanager "github.com/otterize/intents-operator/src/operator/controllers/linkerd"
+	"github.com/otterize/intents-operator/src/shared/injectablerecorder"
+	"github.com/otterize/intents-operator/src/shared/serviceidresolver"
+	"github.com/sirupsen/logrus"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type LinkerdReconciler struct {
+	client.Client
+	Scheme                 *runtime.Scheme
+	RestrictedToNamespaces []string
+	linkerdManager         linkerdmanager.LinkerdPolicyManager
+	serviceIdResolver      serviceidresolver.ServiceResolver
+	injectablerecorder.InjectableRecorder
+}
+
+func NewLinkerdReconciler(c client.Client,
+	s *runtime.Scheme,
+	namespaces []string,
+	enforcementDefaultState,
+	enableLinkerdPolicyCreation bool,
+) *LinkerdReconciler {
+	linkerdreconciler := &LinkerdReconciler{
+		Client:                 c,
+		Scheme:                 s,
+		RestrictedToNamespaces: namespaces,
+		serviceIdResolver:      serviceidresolver.NewResolver(c),
+	}
+
+	linkerdreconciler.linkerdManager = linkerdmanager.NewLinkerdManager(c, namespaces, &linkerdreconciler.InjectableRecorder, enforcementDefaultState, enableLinkerdPolicyCreation)
+	return linkerdreconciler
+}
+
+func (r *LinkerdReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logrus.Info("Starting linkerd reconcile logic")
+
+	installed, err := linkerdmanager.IsLinkerdServerInstalled(ctx, r.Client)
```

Comment:
```suggestion
	installed, err := linkerdmanager.IsLinkerdInstalled(ctx, r.Client)
```
This checks if the Linkerd Server CRD is installed, but in fact it's checking if Linkerd itself is installed in the cluster through this check.